bed/modify.go

```diff
@@ -92,3 +92,178 @@ func MergeHighMem(records []Bed, mergeAdjacent bool) []Bed {
 	outList = append(outList, currentMax)
 	return outList
 }
+
+func mergeKeepLowNameAndScore(records []Bed) (out []Bed, newHidden bool) {
+	var dist int
+	newHidden = false
+	var outList = make([]Bed, 0)
+	SortByCoord(records)
+	var curr = records[0]
+	for i := 1; i < len(records); i++ {
+		if Overlap(curr, records[i]) {
+			if records[i].Score < curr.Score {
+				curr = records[i]
+			}
+		} else if curr.Chrom == records[i].Chrom {
+			dist = records[i].ChromStart - curr.ChromEnd
```

Comment:
Note to self: I think this is where we "use nearest wormhole when it's nearer than bedpe's either foot's nearest gene"
curr is records[i] if it has the lower score
curr.ChromEnd=3 (the last base occupied by curr is 2), records[i].ChromStart=5 (the first base occupied by records[i] is 5). dist = 5-3 = 2 (the distance is length 2, aka base positions 3 and 4)